### ISSUE TITLE
Search : Add search filter to show photos with a specific number of faces (people) in.

### ISSUE THREAD

**Issue by nodecentral** [photoprism/photoprism#3196]
**Describe what problem this solves and why this would be valuable to many users**

It is currently not possible to search for photos with exactly 4 people in , ` faces:4 ` works by showing you all photos with at least 4 faces on it --> https://docs.photoprism.app/user-guide/search/filters/#filter-reference.

As someone who takes a lot of group photos (I’m sure I’m not alone in having loads of wedding, birthdays pictures). it would be great to just be able to locate the ones with specific people in e.g. photos of 4 specific family members together (no one else).

**Describe the solution you'd like**

Have the ability to make the faces and/or person request work exactly as asked, rather than work as just the minimum amount to return. it would be good if ` faces:4 ` worked absolutely and ` faces:4+` was then the call for it to be the minimum amount of faces.

**Describe alternatives you've considered**

**Additional context**

This would be a great filter option if you are seeking to find specific people in

**Comment by kvpp55** [comment]
Who is the most posted person on the internet!?

**Comment by kevincox** [comment]
I agree that I expected `faces:3` to be absolute. It may be nice to add general syntax for order and range operators. For example `faces:>3`. It could also be useful for dates `before:` and `after:` could be unified with `date:>2023-01-01` or similar. If we don't want to do that `faces:=4` could work. Or a separate operator such as `exact-faces:4`.

This could be resolved with negative filters (https://github.com/photoprism/photoprism/issues/2073) in which case it would be `faces:3 -faces:4` to find exactly 3. Not exactly elegant but in many cases the first term may be redundant, so for example you would search `person:"Jane & John" -faces:3`

**Comment by kevincox** [comment]
One other argument for `faces:` being "at least" is that face detection isn't perfect. There are definitely cases where a tree in the background happens to register as a face. So filtering those out may be somewhat surprising.
